EngineLayer/Silac/SilacConversions.cs

```diff
@@ -332,267 +269,602 @@ public static SpectraFileInfo GetHeavyFileInfo(SpectraFileInfo originalFile, Sil
             return new SpectraFileInfo(heavyFileName, originalFile.Condition, originalFile.BiologicalReplicate, originalFile.TechnicalReplicate, originalFile.Fraction);
         }
 
+        public static HashSet<FlashLFQ.ProteinGroup> CleanPastProteinQuant(HashSet<FlashLFQ.ProteinGroup> originalProteinGroups)
+        {
+            HashSet<FlashLFQ.ProteinGroup> cleanedProteinGroups = new HashSet<FlashLFQ.ProteinGroup>();
+            foreach (FlashLFQ.ProteinGroup pg in originalProteinGroups)
+            {
+                cleanedProteinGroups.Add(new FlashLFQ.ProteinGroup(pg.ProteinGroupName, pg.GeneName, pg.Organism));
+            }
+            return cleanedProteinGroups;
+        }
 
         //If SILAC (Post-Quantification), compress the light/heavy protein group pairs into the same light protein group but different files
         //Create new files for each silac label and file so that "file 1" now becomes "file 1 (light)" and "file 1 (heavy)"
         //Change heavy residue into the light residue plus a string label ("PEPTIDEa" -> "PEPTIDEK(+8.014)")
         //This light to heavy conversion needs to happen for the flashLFQ peptides here, but can't for the psm peptides, which are constrained to the protein
         //i.e. pwsms currently don't have sequences; they have start/end residues and a protein sequence. We have to change the output sequences when they're created.
-        public static void SilacConversionsPostQuantification(List<SilacLabel> silacLabels, List<SpectraFileInfo> spectraFileInfo, List<ProteinGroup> ProteinGroups,
-            HashSet<DigestionParams> ListOfDigestionParams, Dictionary<string, List<string>> silacProteinGroupMatcher, FlashLfqResults FlashLfqResults,
-            List<PeptideSpectralMatch> allPsms, Dictionary<string, int> ModsToWriteSelection, bool Integrate)
+        public static void SilacConversionsPostQuantification(List<SilacLabel> allSilacLabels, SilacLabel startLabel, SilacLabel endLabel,
```

Comment:
this method is >500 lines. I see many lines being hit during unit tests, which is excellent. But, I'm not sure if the value returned from subsections are being evaluated. Maybe it's good enough if the final result is accurate. My personal preference would be for this section of code to be broken into isolated methods that could be tested. I don't think there is a policy. Like others to weigh in.